src/memspaces/memspace_highest_bandwidth.c

```diff
@@ -105,7 +105,6 @@ static void umfMemspaceHighestBandwidthInit(void) {
         LOG_DEBUG(
             "Creating the highest bandwidth memspace failed with the error: %u",
             ret);
-        assert(ret == UMF_RESULT_ERROR_NOT_SUPPORTED);
```

Comment:
It can be replaced with: `if (ret != UMF_RESULT_ERROR_NOT_SUPPORTED) LOG_FATAL("...")`

Reply:
Done.